src/dcat_ap_no_validator_service/service/validator_service.py

```diff
@@ -29,42 +26,52 @@ class ValidatorService:
     """Class representing validator service."""
 
     __slots__ = (
-        "url",
-        "data",
-        "shapes",
+        "data_graph",
+        "data_graph_url",
+        "shapes_graph",
+        "shapes_graph_url",
         "config",
-        "ontology",
+        "ontology_graph",
     )
 
     def __init__(
         self,
-        url: Any,
-        data: Any,
-        shapes: Any,
+        data_graph_url: Any,
+        data_graph: Any,
+        shapes_graph: Any,
+        shapes_graph_url: Any,
         config: Config = None,
     ) -> None:
         """Initialize service instance."""
-        self.url = url
-        self.data = fetch_graph(url) if self.url else parse_text(data)
+        self.data_graph_url = data_graph_url
+        self.data_graph = (
+            fetch_graph(data_graph_url)
+            if self.data_graph_url
+            else parse_text(data_graph)
+        )
+        self.shapes_graph_url = shapes_graph_url
+        self.shapes_graph = (
+            fetch_graph(shapes_graph_url)
+            if self.shapes_graph_url
+            else parse_text(shapes_graph)
+        )
         if config is None:
             self.config = Config()
         else:
             self.config = config
-        self.ontology = Graph()
-        self.shapes = shapes
-        if self.shapes is not None:
-            self.shapes = parse_text(shapes)
+        self.ontology_graph = Graph()
 
     async def validate(self) -> Tuple[bool, Graph, Graph, Graph]:
         """Validate function."""
         # Do some sanity checks on preconditions:
-        if len(self.data) == 0:  # No need to validate empty graph
-            raise ValueError("Input graph cannot be empty.")
+        # No need to validate when empty data graph:
+        if self.data_graph is None or len(self.data_graph) == 0:
```

Comment:
Kan det være at det holder med slikt?

```suggestion
        if self.data_graph:
```